app/src/main/java/org/wikipedia/gallery/ImageInfo.kt

```diff
@@ -46,16 +46,25 @@ class ImageInfo {
     val width = 0
     val height = 0
 
-    val bestDerivative get() = derivatives.lastOrNull()
+    fun getBestDerivativeForSize(widthDp: Int): Derivative? {
```

Comment:
Can we add `@Suppress("KotlinConstantConditions")` for this function?

Reply:
of course!